internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
```

Comment:
The function `GenerateSchema` might not handle types that have unexported fields correctly because of the `reflector.Reflect(v)` call. You should ensure that the type `T` has appropriate public fields for jsonschema generation. Consider adding an error handling mechanism if the reflection fails or does not yield a valid schema.